src/internal/connector/exchange/event_calendar_cache.go

```diff
@@ -75,81 +75,55 @@ func (ecc *eventCalendarCache) Populate(
 		return err
 	}
 
-	for _, containerr := range directories {
-		if err := ecc.AddToCache(ctx, containerr); err != nil {
+	for _, container := range directories {
+		if err := checkIDAndName(container); err != nil {
```

Comment:
I am concerned about the use of the `iterateErr` within the wrap method. If a container was unable to be added to the lookup table, would we want to wait until the end of the queries to report it back or would we want to stop the moment  that it was discovered that the lookup was incomplete?

Reply:
>would we want to wait until the end of the queries to report it back or would we want to stop the moment that it was discovered that the lookup was incomplete?

I think the answer to this depends to some extent on what the error policy is. We're not propagating that to the container resolvers at the moment (we may in the future if we feel like it), but if Corso is supposed to be following the "best-effort" policy it feels like it should report the errors but continue anyway?

Reply:
This seems like the correct approach, but that differs from what is actually done:
```go
	resolver, err := exchange.PopulateExchangeContainerResolver(
			ctx,
			qp,
			qp.Scope.Category().PathType(),
		)
		if err != nil {
			return nil, errors.Wrap(err, "getting folder cache")
		}
```

Reply:
so it does. However, fixing the code you cited seems like something that would best be handled in a different PR as it doesn't pertain to consolidating the resolver code. Also delaying returning an error vs. returning an error right away in the resolver code is not going to change the overall outcome since the code you cited will still return early